src/components/Pages/MainPage/MainPage.tsx

```diff
@@ -21,39 +22,42 @@ import { Props } from './MainPage.types'
 import styles from './MainPage.module.css'
 
 function MainPage(props: Props) {
-  const { isLoading, profileAddress, loggedInAddress, isBlocked, isCreationsTabEnabled } = props
+  const { isLoading, profileAddress, loggedInAddress, isBlocked, isCreationsTabEnabled, isAssetsTabEnabled } = props
   const view = getView(loggedInAddress, profileAddress)
   const isMobile = useMobileMediaQuery()
   const navigate = useNavigate()
   const params = useParams()
 
-  if (!isTabValid(params.tab)) {
+  if (
+    !isTabValid(params.tab) ||
+    (params.tab === AccountTabs.ASSETS && !isAssetsTabEnabled) ||
+    (params.tab === AccountTabs.CREATIONS && !isCreationsTabEnabled)
+  ) {
     navigate(locations.account(profileAddress ?? nullAddress), { replace: true })
   }
 
   const selectedTab = useMemo(() => {
     const tab = params.tab
-    switch (tab) {
-      case AccountTabs.CREATIONS:
-        return AccountTabs.CREATIONS
-      default:
-        return AccountTabs.OVERVIEW
-    }
+    return tab || AccountTabs.OVERVIEW
```

Comment:
As it doesn't have a select anymore, what about changing the way we write the selected tab as:
```
const selectedTab = params.tab ?? AccountsTabs.OVERVIEW
```